### PR TITLE
`LazyBuildMixIn.getEstimatedDurationCandidates` duplication

### PR DESCRIPTION
**Pull request by jglick**
@jtnord spotted a mistake in a recent reimplementation of the method used to estimate the duration of new builds: https://github.com/jenkinsci/jenkins/pull/7998#discussion_r1242446688

### Testing done

Looked in vain for any direct test coverage of `Job.getEstimatedDurationCandidates`. Did find test coverage of `Job.getEstimatedDuration`, which is as good (perhaps better, since that is the only caller at least in core of this method). Unfortunately this `SimpleJobTest` used a cumbersome mock system introduced in 40ffa913f8c4e6f644a248c5c943275426aada52, which was amended to run with a live Jenkins server in 1774927de2c87e9f8142649e93901b1340380f59 but still using a mock which did not pick up my `AbstractProject.getEstimatedDurationCandidates` override from #7998. After fixing that, the test still failed to reproduce the problem, because it was too weak: calculated the estimate only after running all builds (when some of the interesting logic occurs only earlier, when there are fewer than three candidates); and also asserting an “average” only when there was only a single value or repeated identical value, never exercising https://github.com/jenkinsci/jenkins/blob/32c51fb7fe05a08a386cf1835b97e33fb65bf3c3/core/src/main/java/hudson/model/Job.java#L1060 outside of these degenerate cases. After introducing logging and strengthening the test, I was able to confirm the bug and its fix.

From what I can tell, the practical impact is pretty small: the more recent of two candidates was being weighted more than the preceding candidate. This is actually a mistake in the direction of my intuition (more recent builds are a better gauge of future behavior), but not consistent or intentional.

### Proposed changelog entries

- Estimate project duration accurately in more cases (regression in 2.407).

### Proposed upgrade guidelines

N/A

### Maintainer checklist

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8233"></a>

### CODE REVIEW

File: test/src/test/java/hudson/model/SimpleJobTest.java
Comment (jglick) [inline test/src/test/java/hudson/model/SimpleJobTest.java:35]: Before fix:

```
java.lang.AssertionError: expected:<18> but was:<17>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at hudson.model.SimpleJobTest.testGetEstimatedDuration(SimpleJobTest.java:35)
```

(expected 15 + 20 / 2, was 15 + 20 + 15 / 3)

File: test/src/test/java/hudson/model/SimpleJobTest.java
Comment (jglick) [inline test/src/test/java/hudson/model/SimpleJobTest.java:79]: Before fix:

```
java.lang.AssertionError: expected:<15> but was:<17>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at hudson.model.SimpleJobTest.testGetEstimatedDurationIfPrevious3BuildsFailed(SimpleJobTest.java:79)
```

(expected 20 + 10 / 2, was 20 + 10 + 20 / 3)

File: test/src/test/java/hudson/model/SimpleJobTest.java
Comment (jglick) [inline test/src/test/java/hudson/model/SimpleJobTest.java:35]: After #7998 but without fix here:

```
java.lang.AssertionError: expected:<175> but was:<167>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at hudson.model.SimpleJobTest.testGetEstimatedDuration(SimpleJobTest.java:35)
```

(expected (200 + 150) / 2 but was ⌈(200 + 150 + 150) / 3⌉)

File: test/src/test/java/hudson/model/SimpleJobTest.java
Comment (jglick) [inline test/src/test/java/hudson/model/SimpleJobTest.java:79]: similarly

```
java.lang.AssertionError: expected:<150> but was:<167>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at hudson.model.SimpleJobTest.testGetEstimatedDurationIfPrevious3BuildsFailed(SimpleJobTest.java:79)
```

(expected (100 + 200) / 2 but was ⌈(100 + 200 + 200) / 3⌉)

**Review by jtnord — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
/label ready-for-merge

---

This PR is now ready for merge. We will merge it after ~24 hours if there is no negative feedback.
Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process.
Thanks!

**Pull request merged**
